### PR TITLE
Allow tranquilizer darts and syringes to stick into people

### PR DESCRIPTION
**Pull request by FlameArrow57**
[FEATURE]

## About the PR 
This PR allows tranquilizer darts and syringes to stick into people, similar to how arrows do. They can be pulled out of your body using the help intent, except for the case of barbed syndicate tranquilizer darts and syringes, which must be cut out with a scalpel. Other than this, no damage is applied when they are in you, so the effects are purely visual.

Throwing darts already could stick into people, but now visual sprites are applied (already existing), and they can also be pulled out with the help intent.

Tranquilizer darts and syringes have new visual sprites for sticking into you, along with new projectile sprites and sprites for the tranquilizer darts when cut out. The sprites for syringes when cut out use existing sprites.

Firing tranquilizer rifle:
https://streamable.com/sc2lzp

Red tranquilizer darts:
image

The existing syringe projectiles were given slight pixel changes, and a barbed syringe variety was made.

Examining someone with darts, syringes, or arrows in them will now show that in the examine report too.

Minor changes to existing implant code and throwing dart code was made to have these effects work.

## Why's this needed? 
It would be cool to have a visual effect to show that you've been hit by one of these projectiles.

## Changelog
```changelog
(u)FlameArrow57
(+)Tranquilizer darts and syringe projectiles can now stick into your body! Along with throwing darts, these items can be pulled out using the help intent, except for the syndicate fired variety, which must be cut out using a scalpel.
```

### CODE REVIEW

File: code/mob/melee_attack_procs.dm [hunk 1]
```diff
@@ -93,6 +116,29 @@
 				src.visible_message("<span class='notice'>[src] pats themselves on the back. Feel better, [src].</span>")
 
 	else
+		var/mob/living/M = target
+
+		var/obj/item/implant/projectile/dart/P = locate(/obj/item/implant/projectile/dart) in M.implant
```
Comment (frawhst) [inline code/mob/melee_attack_procs.dm:121]: any reason syringes and darts don't share a common type ancestor? something like `/obj/item/implant/projectile/body_visible/dart` and `.../syringe`
that way you can check for both at the same time; just include a var in the implant for the pull-out string (`syringe` vs `dart` vs whatever)

also has the bonus of avoiding strange edge cases- as written, if I have a barbed dart and an unbarbed syringe, nobody will be able to pull the syringe out.

File: code/mob/melee_attack_procs.dm [hunk 2]
```diff
@@ -93,6 +116,29 @@
...
+
+		if (P?.barbed == FALSE)
+			src.visible_message("<span class='alert'><B>[src] pulls a dart out of [target]!</B></span>")
+			P.on_remove(M)
+			M.implant.Remove(P)
+			P.set_loc(M.loc)
```
Comment (frawhst) [inline code/mob/melee_attack_procs.dm:127]: would rather this put it in the remover's hand; helps it feel like you're actually pulling it out

File: code/mob/melee_attack_procs.dm [hunk 3]
```diff
@@ -93,6 +116,29 @@
 				src.visible_message("<span class='notice'>[src] pats themselves on the back. Feel better, [src].</span>")
 
 	else
+		var/mob/living/M = target
+
+		var/obj/item/implant/projectile/dart/P = locate(/obj/item/implant/projectile/dart) in M.implant
+
+		if (P?.barbed == FALSE)
+			src.visible_message("<span class='alert'><B>[src] pulls a dart out of [target]!</B></span>")
+			P.on_remove(M)
```
Comment (frawhst) [inline code/mob/melee_attack_procs.dm:125]: I was going to suggest that you add their blood DNA to the implant but as it turns out no bullets do that (????)
gonna add that in a bit, might merge conflict sorry
HOWEVER, most bullets leave a forensic profile on the bullet (which will match the gun when scanned), and these should do that too. Looks like you just need to set `forensic_id` on the implant.
Example:
image

File: code/mob/melee_attack_procs.dm [hunk 4]
```diff
@@ -93,6 +116,29 @@
...
+			P = locate(/obj/item/implant/projectile/syringe) in M.implant
+
+			if (P?.barbed == FALSE)
+				src.visible_message("<span class='alert'><B>[src] pulls a syringe out of [target]!</B></span>")
+				P.on_remove(M)
+				M.implant.Remove(P)
+				P.set_loc(M.loc)
+			else if (P?.barbed == TRUE)
+				src.visible_message("<span class='alert'><B>[src] tries to pull a syringe out of [target], but it's stuck in!</B></span>")
+
```
Comment (frawhst) [inline code/mob/melee_attack_procs.dm:141]: Would like to return after successfully pulling an implant out; feels weird brushing off your shoes every time you pull a dart out of yourself.

File: code/obj/item/implant.dm [hunk 5]
```diff
@@ -855,6 +856,54 @@ THROWING DARTS
...
+		tranq_dart_mutadone
+			name = "tranquilizer dart"
+			desc = "A small tranquilizer dart, emptied of its contents. This one is specialized for removing genetic mutations."
+			icon_state = "tranqdart_green"
+
+			New()
+				..()
+				implant_overlay = image(icon = 'icons/mob/human.dmi', icon_state = "tranqdart_green_stick_[rand(0, 4)]", layer = MOB_EFFECT_LAYER)
+
+	syringe
```
Comment (frawhst) [inline code/obj/item/implant.dm:890]: To a newbie it might be confusing why they can't use the 'syringe' they just found on the floor- a more descriptive name like "spent syringe projectile" or "spent syringe round" would be nice

File: code/obj/item/implant.dm [hunk 6]
```diff
@@ -1689,14 +1744,18 @@ circuitry. As a result neurotoxins can cause massive damage.<BR>
 /* ------------------------- Throwing Darts ---------------------- */
 /* =============================================================== */
 
-/obj/item/implant/projectile/bardart
+/obj/item/implant/projectile/dart/bardart
```
Comment (frawhst) [inline code/obj/item/implant.dm:1747]: In testing these didn't embed into humans- assuming that's a bug since there's code for the overlays

**Review by frawhst — COMMENTED** [review]
Will also want someone to do a pass on the sprites, but they're pretty small so I don't really mind either way

**Comment by frawhst** [comment]
overlays look good 👍

**Comment by frawhst** [comment]
also wow the bleed loop code is grody- individual spawn loops for each implanted projectile. gonna maybe fix that too (on master)

File: code/mob/living/carbon/human/procs/get_desc.dm [hunk 7]
```diff
@@ -129,6 +129,24 @@
 				if (lowertext(sechud_flag) != "none")
 					. += "<br><span class='notice'>[src.name] has a Security HUD flag set:</span> <span class='alert'>[sechud_flag]</span>"
 
+	if (locate(/obj/item/implant/projectile/dart) in src.implant)
+		var/count = 0
+		for (var/obj/item/implant/projectile/dart/P in src.implant)
+			count++
+		. += "<br><span class='alert'>[src] has [count > 1 ? "darts" : "a dart"] in them!</span>"
```
[the remark this answers is not in the document]
Comment (FlameArrow57) [reply code/mob/living/carbon/human/procs/get_desc.dm:136]: I went with "stuck in them" if that's okay.

File: code/mob/melee_attack_procs.dm (hunk 2 above)
Comment (FlameArrow57) [reply code/mob/melee_attack_procs.dm:127]: I tried using the Attackhand proc for items for putting it in hand, seems it plays the sound made when you pick up something, if that works.

File: code/mob/melee_attack_procs.dm (hunk 1 above)
Comment (FlameArrow57) [reply code/mob/melee_attack_procs.dm:121]: No reason in particular, sounds like a better idea

File: code/obj/item/implant.dm (hunk 5 above)
Comment (FlameArrow57) [reply code/obj/item/implant.dm:890]: I didn't put that on the barbed ones, felt maybe it was a bit wordy saying something like spent barbed tranquilizer dart/syringe, let me know if I should. I tried to add a bit more to the descriptions for those to make up for it.

File: code/obj/item/implant.dm (hunk 6 above)
Comment (FlameArrow57) [reply code/obj/item/implant.dm]: Bar darts have a 5% chance to stick into you, which I left that way. Maybe that was the reason? Changing it to 100%, seemed to work fine for me. Not sure otherwise

**Comment by Tarmunora** [comment]
🤔 might be interesting if tranq darts did a gradual release of chems over time with this, so if you got tranqed and could buy some space/time to pull them out, you can counterplay them a bit better. Also works well with the syndicate ones being unable to be removed by hand

**Comment by frawhst** [comment]
!merge_upstream
